### ISSUE TITLE
How should we treat "spoilers"?

### ISSUE THREAD

**Issue by kellhus** [kellhus/munkres-study-notes#2]
Should we refrain from posting our solutions until a specific deadline? (This might not be feasible -- there can be issues with conflicting commits if everyone tries to submit things on the same day and has no idea what others have written in the meantime.)

Or should we let people be responsible for not looking at the solutions until they are ready? (And update the repository continuously without a specific deadline in mind. This better approximates the typical forum or reddit-type-website experience.)

AddemF and others, what do you think? The `Academic Integrity` subsection of the `README.md` needs to reflect the decision.

**Comment by AddemF** [comment]
Yeah, I'm in favor of the latter.  If you don't fully know the solution yet and want just a little push but not the full answer, you can communicate in the IRC or on Reddit, or whatever.  Only start posting your answer or reading other answers when you feel you have it figured out.

**Comment by pqnelson** [comment]
I actually do not know...because, ostensibly, I'd like to keep my solutions separate from someone else's. Not because I'm antisocial, but I don't want to get confused...And I'd like feedback on what I did well, or badly, or confusingly, or elegantly or...etc.

What is the current designation for placing solutions? Does everyone put it in a separate file? Or keep it locally? Or place it in the notes directly?

**Comment by AddemF** [comment]
This is just my conception of how our organization is evolving here, but it seems like this GitHub project is more of a polished object, or like a "finished project" (that isn't quite finished yet).  It would make sense to me if we used Reddit as the kind of interactive, messy conversation where we post our attempts at solutions and have each other give feedback and so on.  Then when we feel good about finalizing what we've done, we publish it here.

**Comment by Thimoteus** [comment]
I imagine we could keep a separate folder for each person, or even a branch for each person.

**Comment by kellhus** [comment]
I propose using the issue tracker to create issues for exercises.

Grossly overestimating, Munkres contains at most $14 \times 6 \times 10 = 840$ exercises (chapters times sections per chapter times exercises per section). Compare that to more than 4,000 issues on the issue tracker of the [IPython / Jupyter project](https://github.com/ipython/ipython/issues). Hence the number of issues should be well within the manageable range.

Just create issues named "Exercise 2.10.4", label them "question" -- or should we create a separate "exercise" label for exercises? probably not, we'll want to edit both discussion and solutions into the final set of notes -- and if you are sure everyone has discussed it and the solution is correct, go ahead and edit the main markdown files.
